client/session.go

```diff
@@ -2344,23 +2479,8 @@ func (s *session) selectBlocksForSeriesFromPeerBlocksMetadata(
 		}
 
 		if len(currEligible) == 0 {
-			// No current eligible peers to select from
-			finalError := true
-			if currUnselected.reattempt.failAllowed != nil && atomic.AddInt32(currUnselected.reattempt.failAllowed, -1) > 0 {
-				// Some peers may still return results so we don't report error here
-				finalError = false
-			}
-
-			if finalError {
-				m.fetchBlockFinalError.Inc(1)
-				s.log.WithFields(
-					xlog.NewField("id", currID.String()),
-					xlog.NewField("start", earliestStart),
-					xlog.NewField("attempted", currUnselected.reattempt.attempt),
-					xlog.NewField("attemptErrs", xerrors.Errors(currUnselected.reattempt.errs).Error()),
-				).Error("retries failed for streaming blocks from peers")
-			}
-
+			// No current eligible peers to select from, retry all of them again
+			currEligible = currStart[:]
```

Comment:
could you emit a metric/log indicating this is happening, will be useful if we ever see bootstrapping stuck for a while.

Reply:
yeah both a log and metric would be useful here just so its easy to debug whats wrong both ways

Reply:
To clarify, if all of our peers error out enough to cause "final errors" this will just restart the retry loop again, right?

Could you add a comment to that effect, this method is pretty confusing if you've never worked with it before.

Reply:
Yeah that's the idea, I'll add a comment to that effect and the metric.  I'll also add an ability to cut this circuit if we need to (i.e. for test cluster).